Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45323&attempt_id=53a3a79f-da27-4314-bdaf-d1f8bdf7523e)

Attempted: ydiff

<details><summary>Partial log (click to expand)</summary><p>

```
---------------------------------------------------
tests/test_ydiff.py     427     34    92%   80, 258-293, 800-802, 821-828, 831-835, 838-842
ydiff.py                482     96    80%   35-37, 319, 322-326, 329-337, 361-362, 401, 407, 481, 560-561, 568-576, 634-655, 718-734, 739-743, 748-750, 756-758, 770-773, 780-790, 814-815, 861-863, 868-872, 874-879, 890-903, 907
---------------------------------------------------
TOTAL                   909    130    86%
PYTHON=python3 tests/regression.sh
/bin/sh: tests/regression.sh: not found
make: *** [Makefile:45: reg3] Error 127
builder for '/nix/store/2f4p2zfv551nyl8pcwkxfqdm16zb0v15-ydiff-1.1.drv' failed with exit code 2
error: build of '/nix/store/2f4p2zfv551nyl8pcwkxfqdm16zb0v15-ydiff-1.1.drv' failed
```
</p></details>

--- a/pkgs/development/tools/ydiff/default.nix
+++ b/pkgs/development/tools/ydiff/default.nix
@@ -1,0 +1,45 @@
+{ stdenv, lib, pythonPackages, python3Packages, less, patchutils, git
+, subversion, coreutils, which }:
+
+with pythonPackages;
+
+buildPythonApplication rec {
+  pname = "ydiff";
+  version = "1.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0mxcl17sx1d4vaw22ammnnn3y19mm7r6ljbarcjzi519klz26bnf";
+  };
+
+  patchPhase = ''
+    substituteInPlace tests/test_ydiff.py \
+      --replace /bin/rm ${coreutils}/bin/rm \
+      --replace /bin/sh ${stdenv.shell}
+    substituteInPlace Makefile \
+      --replace "pep8 --ignore" "# pep8 --ignore" \
+      --replace "python3 \`which coverage\`" "${python3Packages.coverage}/bin/coverage3" \
+      --replace /bin/sh ${stdenv.shell} \
+      --replace tests/regression.sh "${stdenv.shell} tests/regression.sh"
+    patchShebangs tests/*.sh
+  '';
+
+  buildInputs = [ docutils pygments ];
+  propagatedBuildInputs = [ less patchutils ];
+  checkInputs = [ coverage coreutils git subversion which ];
+
+  checkTarget = if isPy3k then "test3" else "test";
+
+  meta = {
+    homepage = https://github.com/ymattw/ydiff;
+    description = "View colored, incremental diff in workspace or from stdin";
+    longDescription = ''
+      Term based tool to view colored, incremental diff in a version
+      controlled workspace (supports Git, Mercurial, Perforce and Svn
+      so far) or from stdin, with side by side (similar to diff -y)
+      and auto pager support.
+    '';
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ leenaars ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -19416,6 +19416,8 @@ with pkgs;
 
   yate = callPackage ../applications/misc/yate { };
 
+  ydiff = callPackage ../development/tools/ydiff { };
+
   yed = callPackage ../applications/graphics/yed {};
 
   inherit (gnome3) yelp;

